tests/src/python/test_provider_postgres.py

```diff
@@ -99,6 +99,29 @@ def execSQLCommand(self, sql):
         cur.close()
         self.con.commit()
 
+    # Create instances of this class for scoped backups,
+    # example:
+    #
+    #  backup1 = self.scopedTableBackup('qgis_test', 'datatable1')
+    #  backup2 = self.scopedTableBackup('qgis_test', 'datatable2')
+    #
+    def scopedTableBackup(self, schema, table):
+
+        class ScopedBackup():
+            def __init__(self, tester, schema, table):
+                self.schema = schema
+                self.table = table
+                self.tester = tester
+                tester.execSQLCommand('DROP TABLE IF EXISTS {s}.{t}_edit CASCADE'.format(s=schema, t=table))
+                tester.execSQLCommand('CREATE TABLE {s}.{t}_edit AS SELECT * FROM {s}.{t}'.format(s=schema, t=table))
+
+            def __del__(self):
+                self.tester.execSQLCommand('TRUNCATE TABLE {s}.{t}'.format(s=self.schema, t=self.table))
+                self.tester.execSQLCommand('INSERT INTO {s}.{t} SELECT * FROM {s}.{t}_edit'.format(s=self.schema, t=self.table))
+                self.tester.execSQLCommand('DROP TABLE {s}.{t}_edit'.format(s=self.schema, t=self.table))
+
+        return ScopedBackup(self, schema, table)
```

Comment:
I like this approach! I can't help but feel like it's only a few tweaks away from being a real context manager though. E.g. it would be cleaner/more pythonic to be able to directly call:

``` 
    def my_test(self):
        # ...
        with self.backup_table('schema','my_table'):
            # some sql

        # ...
```

Reply:
I thought about that but then some tests use more than a single table, so the `with` would need multiple tables to be backed up